src/index.js

```diff
@@ -97,8 +100,24 @@ export default class ESjs {
     }
 
     this.ensureFieldIndex(field);
-    this.addTokens(doc, field);
-    this.addTerms(doc, field);
+
+    let analyzer = 'standard';
+
+    if (this.fields[field] && this.fields[field].analyzer) {
+      analyzer = this.fields[field].analyzer;
+    }
+
+    switch (analyzer) {
```

Comment:
Now that we're validating, the default should be impossible to reach, which means we could alternatively do:

```
const operations = {
  keyword: [this.addTerms],
  standard: [this.addTokens, this.addTerms],
};

operations[analyzer].forEach(op => op.apply(...))
```

which makes it easier to extend, IMO.

Reply:
Good call, forgot to come back around to this piece.